### ISSUE TITLE
Sekiro: horrible framedrops and stuttering especially if refreshrate is above 60

### ISSUE THREAD

**Issue by Yasand123** [doitsujin/dxvk#3092]
Framerate drops to the 50s and 40s and jumps all over the place with unstable frametimes when rotating the camera, or moving from one place to another. These stutters and framedrops aren't related to shader caching from what I can tell. It probably has to do with asset streaming.

Keeping the monitor's refreshrate at 60 helps a lot, but doesn't completely eliminate the problem.

Difference between 75Hz and 60 Hz is quite interesting. If your refreshrate is higher than 60, whenever the framerate drops it will stay dropped. On the other hand, framedropping in 60Hz is temporary, as it will go back to normal in seconds. It is still very annoying though in many other areas, rotating the camera can be nauseating in both cases.

Here's a video demonstrating the 2 issues:
https://streamable.com/zhbq4q

It is worth noting that I'm not the only one experiencing this, other reports are saying the same thing over on [ProtonDB](https://www.protondb.com/app/814380), in fact I got the keeping Hz at 60 idea from there.

### Software information
Sekiro on Steam. Tried highest and lowest graphical settings, same behavior.

### System information
- OS: Arch Linux 6.0.9
- GPU: NVIDIA RTX 2060 SUPER
- Driver: 520.56.06
- Wine version: Proton experimental 7.0-20221115 -- this has been happening since I bought the game 7 months ago.
- DXVK version: 2.0.0 -- also same as above.

### Log files
I want to provide logs but I am not sure where they are located as I'm running the game from steam.

**Comment by Blisto91** [comment]
> I want to provide logs but I am not sure where they are located as I'm running the game from steam.

If you launch the game with `PROTON_LOG=1 %command%` the dxvk logs should appear next to the games exe or in it's root game folder.

Also if the game allows it then try to make a apitrace https://github.com/doitsujin/dxvk/wiki/Using-Apitrace

**Comment by K0bin** [comment]
I'm fairly sure this isn't a DXVK problem. I had the same problem when I played the game so I dediced to try it with DXVK on Windows and it was perfectly fine.

**Comment by Ryusennin** [comment]
You should note that Sekiro is hard-locked to 60 fps. Forcing it to run above 60 Hz will introduce stuttering, unless your monitor supports VRR (although at 144 Hz VRR, the game will still be capped at 60 fps). On Windows, dual monitors are also known to cause severe performance issues in this game.

I think there's a Linux version of Sekiro FPS Unlocker, maybe it's worth a try.

**Comment by doitsujin** [comment]
Is this with vsync enabled? I can certainly see things getting really wonky with 75Hz + Vsync + the game's barely functioning FPS cap, but I'm not sure if this is really that much better on Windows, what's causing this, or how to fix it without breaking a million other things. Timing-related issues tend to be extremely complicated.

As for other kinds of stutter in certain parts of the game, yes, that is normal. I do not know why this only happens with DXVK, but I've already spent a significant amount of time trying to debug that, and the only thing I found is that there's some really slow memcpy somewhere that's not being called from our code and no existing DXVK workaround helps with that.

**Comment by mbriar** [comment]
@Yasand123 maybe try and see if it's better with latest master.

**Comment by Yasand123** [comment]
> @Yasand123 maybe try and see if it's better with latest master.
@mbriar 
I tried that at the time and didn't notice any improvements. I tried the latest update that was released an hour ago, and that completely fixed the issues. I don't know if I did something wrong while testing it a week ago or whether this was only fixed recently thanks to this https://github.com/doitsujin/dxvk/compare/0af5ececa6ba250b529c9eb988791cf4c0a347e5...master

Currently, there's odd visual jitters as I rotate the camera, but this is completely unrelated to frametimes, frametimes are _impressively_ stable now. The jitters are gone when increasing the game's framerate to match the monitor's refreshrate, with something like [this](https://github.com/Lahvuun/sekirofpsunlock). Limiting refreshrate to 60 also eliminates the visual jitter problem. If I had to describe this issue, it's exactly like pan jittering in anime or movies.

@doitsujin @Blisto91 @K0bin but yeah it runs buttery smooth now on my machine. Thank you all for your hard work! This should be closed now I think.
